opendevin/core/logger.py

```diff
@@ -71,13 +72,22 @@ def format(self, record):
 llm_formatter = logging.Formatter('%(message)s')
 
 
+class SensitiveDataFilter(logging.Filter):
+    def filter(self, record):
+        api_key_pattern = r'sk-\w+'
```

Comment:
The OpenAI keys can look like that, `"sk-BvM..."`, and the newer keys look like api_key=`"sk-proj-BvM..."`. Can we tweak this pattern a bit, to capture the second word?

Actually, Anthropic keys look like `"sk-ant-api04-something-something_else-another_word-another_word"`...